lib/UA.js

```diff
@@ -129,7 +120,17 @@ function UA(uaString) {
 	if (normalized) {
 		this.ua = new useragent.Agent(normalized[1], normalized[2], (normalized[3] || 0), (normalized[4] || 0));
 	} else {
-		this.ua = useragent.parse(stripiOSWebViewBrowsers(uaString));
+
+		// Chrome and Opera on iOS uses a UIWebView of the underlying platform to render
+		// content, by stripping the CriOS or OPiOS strings the useragent parser will alias the
+		// user agent to ios_saf for the UIWebView, which is closer to the actual renderer
+		uaString = uaString.replace(/((CriOS|OPiOS)\/(\d+)\.(\d+)\.(\d+)\.(\d+)|(FxiOS\/(\d+)\.(\d+)))/, '');
+
+		// Vivaldi browser is recognised by UA module but is actually identical to Chrome, so
+		// the best way to get accurate targeting is to remove the vivaldi token from the UA
+		uaString = uaString.replace(/ vivaldi\/[\d\.]+\d+/i, '');
```

Comment:
Is this possible to do using our function based aliases? https://github.com/Financial-Times/polyfill-service/blob/master/lib/UA.js#L86